src/plugins/intel_cpu/src/transformations/cpu_opset/common/op/rope.hpp

```diff
@@ -0,0 +1,53 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <ngraph/node.hpp>
+#include <ngraph/op/op.hpp>
+
+namespace ov {
+namespace intel_cpu {
+
+class RoPENode : public ngraph::op::Op {
+public:
+    OPENVINO_OP("RoPE", "cpu_plugin_opset");
+
+    RoPENode() = default;
+
+    struct Config {
+        size_t slice_start = 0;  // slice inner-most dimensions of input
+        size_t slice_stop = 0;
+        bool input_trans0213 = false;  // transpose input dim 1&2
+                                       //   from [batch, seq_length, head_cnt, head_dims]
+                                       //   into [batch, head_cnt, seq_length, head_dims]
+                                       //   before RoPE
+        bool is_interleaved = false;   // interleaved mode (for GPTJ)
+                                       //   this also implies trans0213 happens after RoPE
+        size_t ndims = 0;
```

Comment:
not all dimension are used for rotary embedding in some LLM, sometime only partial are used and the rest dimensions are just copied, to avoid confusion I changed the name into `rotary_ndims` and added a comment.